pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -393,17 +393,18 @@ public IdealState apply(IdealState idealState) {
     updateIdealState(helixManager, tableName, updater, DEFAULT_RETRY_POLICY);
   }
 
-  public static List<String> getEnabledInstancesWithTag(HelixAdmin helixAdmin, String helixClusterName, String instanceTag) {
+  public static List<String> getEnabledInstancesWithTag(HelixAdmin helixAdmin, String helixClusterName,
+      String instanceTag) {
     List<String> instances = helixAdmin.getInstancesInCluster(helixClusterName);
     List<String> enabledInstances = new ArrayList<>();
     for (String instance : instances) {
-      InstanceConfig instanceConfig = helixAdmin.getInstanceConfig(helixClusterName, instance);
-      if (instanceConfig == null) {
-        LOGGER.warn("InstanceConfig not found for instance: {}", instance);
-        continue;
-      }
-      if (instanceConfig.containsTag(instanceTag) && instanceConfig.getInstanceEnabled()) {
-        enabledInstances.add(instance);
+      try {
+        InstanceConfig instanceConfig = helixAdmin.getInstanceConfig(helixClusterName, instance);
```

Comment:
It can never be null. Helix will directly throw exception if instance does not exist.